src/bio/bio_wal.c

```diff
@@ -1787,6 +1812,23 @@ bio_wal_replay(struct bio_meta_context *mc,
 	return rc;
 }
 
+int
+bio_wal_replay_stats(struct bio_meta_context *mc, struct bio_wal_rp_stats **wrsp,
+		     int (*replay_cb)(uint64_t tx_id, struct umem_action *act, void *arg),
+		     void *arg)
+{
+	int rc = bio_wal_replay(mc, replay_cb, arg);
+
+	/* Return WAL replay metrics */
+	if (rc >= 0) {
+		struct bio_wal_rp_stats *wrs = mc->mc_wr_stats;
+
+		if (wrs)
+			*wrsp = wrs;
+	}
+	return rc;
+}
```

Comment:
Why don't we just change bio_wal_replay() to accept an optional "stats"? If passed in 'stats' is non-NULL, bio_wal_reply() fills stats on return.